### ISSUE TITLE
Generic.WhiteSpace.ScopeIndent reports incorrect errors when indenting double arrows in short arrays

### ISSUE THREAD

**Issue by dpauli** [squizlabs/PHP_CodeSniffer#1465]
I found a very strange behaviour with line endings in an array of arrays. Lets look to the following code.

# Correct code
```php
<?php
declare(strict_types=1);
return [
    'veryLongKeySoIWantToMakeALineBreak' => 'veryLonValueSoIWantToMakeALineBreak',

    'someOtherKey' => [
        'someValue'
    ],

    'arrayWithArraysInThere' => [
        ['Value1', 'Value1']
    ],
];
```
There is no error, no problem, no warning with CodeSniffer.

After a line break it also looks good without any problem

```php
<?php
declare(strict_types=1);
return [
    'veryLongKeySoIWantToMakeALineBreak'
    => 'veryLonValueSoIWantToMakeALineBreak',

    'someOtherKey' => [
        'someValue'
    ],

    'arrayWithArraysInThere' => [
        ['Value1', 'Value1']
    ],
];
```

# Lets intend after the line break

Because of a not good looking intendation of first key/value assignment I move the assignment with 4 spaces
```php
<?php
declare(strict_types=1);
return [
    'veryLongKeySoIWantToMakeALineBreak'
        => 'veryLonValueSoIWantToMakeALineBreak',

    'someOtherKey' => [
        'someValue'
    ],

    'arrayWithArraysInThere' => [
        ['Value1', 'Value1']
    ],
];
```
No there is the first warning: **phpcs: Line intended incorrectly; expected at least 8 spaces, found 4**. But I get this warning on the last key/value assignment (on *arrayWithArrayInThere*). Why I get this?

# "Solve" this problem with the third assignment

This warning can be solved, not only be intending the first assignment but changing the third key/value pair.

```php
<?php
declare(strict_types=1);
return [
    'veryLongKeySoIWantToMakeALineBreak'
        => 'veryLonValueSoIWantToMakeALineBreak',

    'someOtherKey' => ['someValue'],

    'arrayWithArraysInThere' => [
        ['Value1', 'Value1']
    ],
];
```
The intendation does not change, only the postion of the value of the third key/value pair. Why? Oo

**Comment by gsherwood** [comment]
Debug output:
```
$ phpcs temp.php --standard=Generic --sniffs=Generic.WhiteSpace.ScopeIndent --runtime-set scope_indent_debug 1
Start with token 0 on line 1 with indent 0
Closing parenthesis found on line 2
	 * ignoring single-line definition *
Closing short array bracket found on line 9
	* first token on line 7 is 24 (T_CONSTANT_ENCAPSED_STRING) *
	* previous is T_CONSTANT_ENCAPSED_STRING on line 5 *
	* amended first token is 17 (T_DOUBLE_ARROW) on line 5 *
	=> checking indent of 4; main indent set to 8 by token 17 (T_DOUBLE_ARROW)
[Line 11] Line indented incorrectly; expected at least 8 spaces, found 4
	=> Add adjustment of 4 for token 39 (T_CONSTANT_ENCAPSED_STRING) on line 11
Closing short array bracket found on line 12
	 * ignoring single-line definition *
Closing short array bracket found on line 13
	* first token on line 11 is 39 (T_CONSTANT_ENCAPSED_STRING) *
	* previous is T_OPEN_SHORT_ARRAY on line 7 *
	* amended first token is 17 (T_DOUBLE_ARROW) on line 5 *
	=> checking indent of 8; main indent set to 8 by token 17 (T_DOUBLE_ARROW)
[Line 13] Line indented incorrectly; expected at least 8 spaces, found 4
	=> Add adjustment of 4 for token 54 (T_CLOSE_SHORT_ARRAY) on line 13
Closing short array bracket found on line 14
	* first token on line 3 is 9 (T_RETURN) *
	=> checking indent of 0; main indent set to 0 by token 9 (T_RETURN)
```
Problem appears to be where scope indent is amended and finds the double arrow on line 5. Looks like a bug in the sniff, but needs investigation.

**Comment by gsherwood** [comment]
Double arrow needed to be ignored (just like comma is) when searching backwards in short arrays. Thanks for the bug report.
